core/domain/classifier_domain.py

```diff
@@ -80,3 +82,45 @@ def to_dict(self):
             'cached_classifier_data': self.cached_classifier_data,
             'data_schema_version': self.data_schema_version
         }
+
+    def validate(self):
+        """Validates the classifier before it is saved to storage."""
+
+        if not isinstance(self.id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.id)
+        utils.require_valid_name(
+            self.id, 'the classifier id')
+
+        if not isinstance(self.exp_id, basestring):
+            raise utils.ValidationError(
+                'Expected exp_id to be a string, received %s' % self.exp_id)
+        utils.require_valid_name(
+            self.exp_id, 'the exploration id')
+
+        if not isinstance(self.exp_version_when_created, int):
+            raise utils.ValidationError(
+                ('Expected exp_version_when_created to be a integer,') + (
+                    'received %d' % self.exp_version_when_created))
+
+        if not isinstance(self.state_name, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.state_name)
+        utils.require_valid_name(
+            self.state_name, 'the state name')
+
+        if not isinstance(self.algorithm_id, basestring):
+            raise utils.ValidationError(
+                'Expected algorithm_id to be a string, received %s' %(
+                    self.algorithm_id))
+        utils.require_valid_name(
+            self.algorithm_id, 'the algorithm id')
+        if (self.algorithm_id != (
+                feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])):
+            raise utils.ValidationError(
+                'Invalid algorithm_id: %s' % self.algorithm_id)
+
+        if not isinstance(self.cached_classifier_data, dict):
```

Comment:
@anmolshkl @seanlip There is no specification for cached_classifier_data anywhere. Right now, there is only LDAStringClassifier. So, I have to create a validate method in LDAStringClassifier which will validate the parameters produced by the classifier? Correct me if I am wrong.